### ISSUE TITLE
ec2 node provisioner not creating nodes when provisioned on rancher using self-signed certs / rancher generated certs.

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#32785]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
* deploy downstream RKE2 cluster using EC2 node provisioner

**Result:**
* ec2 does not create any nodes when using the provisioner -> rancher never registers the cluster

**Other details that may be helpful:**

Only 1 use case working: 
HA Rancher server with lets encrypt, provisioning 1 node cluster with all roles 

Rancher Setup Use Cases Broken:
* HA with Rancher generated certs
* HA with BYO valid certs
* Docker Install 

here are some rancher logs while provisioning an ec2 cluster :
[test2.txt](https://github.com/rancher/rancher/files/6517933/test2.txt)

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 
- Installation option (single install/HA): master-head (55f0bb2)

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): hosted AWS/custom AWS nodes
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`): v1.21.1-alpha4+rke2r1

**Comment by deniseschannon** [comment]
@slickwarren Are you saying this is broken with multi-node EC2 clusters? If so, it might be the same issue as multi-node clusters and requiring an extra port now.

**Comment by slickwarren** [comment]
this ticket is for single node clusters at the moment. We are blocked by this [separate issue for multi-node clusters on EC2](https://github.com/rancher/rancher/issues/32779#issue-894730155) @deniseschannon

**Comment by slickwarren** [comment]
on master-head (4f9850d):
rancher installed via docker (single node) or installed on HA setup result in the same error -- downstream cluster using EC2 to provision:
```
failed to create fleet-default/cw-aws-ec2provisioned cluster.x-k8s.io/v1alpha4, Kind=Cluster for rke-cluster fleet-default/cw-aws-ec2provisioned: Internal error occurred: failed calling webhook "default.cluster.cluster.x-k8s.io": Post "https://webhook-service.cattle-system.svc:443/mutate-cluster-x-k8s-io-v1alpha4-cluster?timeout=10s": dial tcp <internal_ip>:443: connect: connection refused, failed to create fleet-default/cw-aws-ec2provisioned-nodepool-cw-hads-all1 cluster.x-k8s.io/v1alpha4, Kind=MachineDeployment for rke-cluster fleet-default/cw-aws-ec2provisioned: Internal error occurred: failed calling webhook "default.machinedeployment.cluster.x-k8s.io": Post "https://webhook-service.cattle-system.svc:443/mutate-cluster-x-k8s-io-v1alpha4-machinedeployment?timeout=10s": dial tcp <internal_ip>:443: connect: connection refused

```

**Comment by kinarashah** [comment]
`Error creating machine: Error in driver during machine creation: Error launching instance: InvalidParameterValue: User data is limited to 16384 bytes\n\tstatus code: 400, request id: 2f755444-143c-476a-b6e6-f0ab18f8c414\nThe default lines below are for a sh/bash shell, you can specify the shell you're using, with the --shell flag."`

**Comment by deniseschannon** [comment]
Letsencrypt uses a public CA versus Rancher generated or BYO, which might be a difference.

The CA checksum logic is invoked in the Rancher generated and BYO, which *could* be the issue, but not confirmed.

**Comment by slickwarren** [comment]
note: for a custom cluster, I was able to provision a single node setup using an ec2 node on master-head (b2f8824). However, the node provisioner for single node still did not work. For custom node with dedicated roles, I got this error using ec2 nodes (from the node that I ran the etcd registration cmd on):
```
time="2021-06-07T21:03:59Z" level=warning msg="Cluster CA certificate is not
 trusted by the host CA bundle, but the token does not include a CA hash. Use the full token from the server's node-toke
n file to enable Cluster CA validation."
```

**Comment by kinarashah** [comment]
Available to test once https://drone-publish.rancher.io/rancher/rancher/6028 passes.

**Comment by slickwarren** [comment]
on master-head (01665d0):
* provision downstream cluster using ec2 provisioner, single node, rancher installed via docker (self signed certs) -- pass
*  provision downstream cluster using ec2 provisioner, single node, rancher is HA installed (byo certs) -- pass
